### PR TITLE
feat(Logo): component refactoring

### PR DESCRIPTION
**Pull request by kendoow**

### CODE REVIEW

File: src/components/Logo/Logo.tsx [hunk 1]
```diff
@@ -31,17 +31,13 @@ export const Logo: React.FC<
...
-    } else if (icon) {
-        buttonIcon = <Icon data={icon} size={iconSize} className={iconClassName} />;
-    }
+    const buttonIcon = iconSrc ? (
+        <Button.Icon className={iconClassName}>
+            <img alt="logo icon" src={iconSrc} width={iconSize} height={iconSize} />
+        </Button.Icon>
+    ) : (
+        icon && <Icon data={icon} size={iconSize} className={iconClassName} />
+    );
```
Comment (Lunory) [inline src/components/Logo/Logo.tsx:40]: Nested condition in ternary operator is difficult for understanding
I prefer to keep old code with `if..else if`

File: src/components/Logo/Logo.tsx [hunk 2]
```diff
@@ -62,37 +58,39 @@ export const Logo: React.FC<
...
-    }
+    const logo = (
+        <div className={b('logo')} style={{fontSize: textSize}}>
+            {logoContent}
+        </div>
+    );
+
+    const renderLogoSection = () => {
+        if (compact) return null;
+        if (hasWrapper && !href) {
```
Comment (Lunory) [inline src/components/Logo/Logo.tsx:71]: `!href` seems like useless check. Wrapper depends on `hasWrapper` prop only

File: src/components/Logo/Logo.tsx (hunk 1 above)
Comment (kendoow) [reply src/components/Logo/Logo.tsx]: fixed

File: src/components/Logo/Logo.tsx (hunk 2 above)
Comment (kendoow) [reply src/components/Logo/Logo.tsx:75]: fixed

**Review by Lunory — APPROVED** [review]

**Comment by Lunory** [comment]
PR is waiting for next major

**Pull request closed without merging**